orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/pipeline/ManualJudgmentStage.groovy

```diff
@@ -72,21 +75,35 @@ class ManualJudgmentStage implements StageDefinitionBuilder, AuthenticatedStage
     final long backoffPeriod = 15000
     final long timeout = TimeUnit.DAYS.toMillis(3)
 
-    @Autowired(required = false)
-    EchoService echoService
+    private final EchoService echoService
+
+    private final FiatPermissionEvaluator fiatPermissionEvaluator
+
+    WaitForManualJudgmentTask(Optional<EchoService> echoService, Optional<FiatPermissionEvaluator> fpe) {
+      this.echoService = echoService.orElse(null)
+      this.fiatPermissionEvaluator = fpe.orElse(null)
+    }
 
     @Override
     TaskResult execute(StageExecution stage) {
       StageData stageData = stage.mapTo(StageData)
+      def stageAuthorized = stage.context.get('isAuthorized')
+      def stageRoles = stage.context.get('stageRoles')
+      def permissions = stage.context.get('permissions')
+      def username = stage.lastModified? stage.lastModified.user: stage.context.get('username')
```

Comment:
I think you should be able to fetch the user from `AuthenticatedRequest.getSpinnakerUser()`.

Defaulting to `stage.context.get('username')` seems suspicious in this case.

Reply:
I should have deleted it. Kept it for executing the test cases. Modified the test case.

Reply:
I still don't believe accessing `stage.lastModified` directly is correct, what does AuthenticatedRequest.getSpinnakerUser() return? 

I _believe_ it should return the correct user.

Reply:
stage.lastModified returns the user, who clicked on the 'continue' or 'stop' button of the manual judgment stage.

this should work. it returns the correct user. 

If you still insist, i can change the code.

Reply:
Yes, `lastModified` is an implementation detail. My preference would be to avoid referencing it directly.

Reply:
done...fetching user from AuthenticatedRequest.getSpinnakerUser().